File: picture_library/src/main/java/com/luck/picture/lib/tools/ToastUtils.java
package com.luck.picture.lib.tools;

import android.content.Context;
import android.widget.Toast;

/**
 * @author：luck
 * @data：2018/3/28 下午4:10
 * @描述: Toast工具类
 */

public final class ToastUtils {
    public static void s(Context context, String s) {
        if (!isShowToast()) {
            Toast.makeText(context.getApplicationContext(), s, Toast.LENGTH_SHORT)
                    .show();
        }
    }

    /**
     * Prevent continuous click, jump two pages
     */
    private static long lastToastTime;
    private final static long TIME = 1500;

    public static boolean isShowToast() {
        long time = System.currentTimeMillis();
        if (time - lastToastTime < TIME) {
            return true;
        }
        lastToastTime = time;
        return false;
    }
}

File: selector/src/main/java/com/luck/picture/lib/utils/ToastUtils.kt
package com.luck.picture.lib.utils

import android.content.Context
import android.widget.Toast
import com.luck.picture.lib.helper.ActivityCompatHelper

/**
 * @author：luck
 * @date：2019-07-17 15:12
 * @describe：Toast Utils
 */
object ToastUtils {
    fun showMsg(context: Context, msg: String) {
        if (!ActivityCompatHelper.assertValidRequest(context)) {
            return
        }
        Toast.makeText(context.applicationContext, msg, Toast.LENGTH_SHORT).show()
    }
}